### PR TITLE
Mesh_3 - document CGAL::remove_far_points_in_mesh_3()

### LINKED ISSUE #5846: "Ghost vertices" in Mesh_3
Hello everyone,
while using CGAL I came across something that might be a bug.

## Issue Details

The tetrahedral mesh created with Mesh_3 sometimes contains vertices far outside the domain that do not seem to belong to any mesh element. This problem appears (at least in my experience) if and only if ```CGAL::Parallel_tag``` is used as a template parameter of ```CGAL::Mesh_triangulation_3```. In my view, this counts as a wrong result.

## Source Code

The problem can be reproduced in the CGAL example ```Mesh_3/mesh_implicit_sphere.cpp```. For simplicity, I compile directly with
```
> g++ -O3 -frounding-math -DCGAL_CONCURRENT_MESH_3 -DCGAL_LINKED_WITH_TBB -DCGAL_HEADER_ONLY -isystem /path/to/eigen -isystem /path/to/CGAL-5.2.1/include -isystem /path/to/boost_1_74_0/include -isystem /path/to/tbb/include -L/path/to/boost_1_74_0/lib/shared -L/path/to/tbb/lib/ -Wl,-rpath,/path/to/boost_1_74_0/lib/shared -Wl,-rpath,/path/to/tbb/lib/ /path/to/CGAL-5.2/examples/Mesh_3/mesh_implicit_sphere.cpp -lgmp -lmpfr -lboost_program_options -lboost_thread -ltbb -ltbbmalloc -lpthread
```
The ghost vertices unfortunately are _not_ visible in medit. However, one can notice that in ```out.mesh``` there are points with such big coordinates that they clearly do not belong to a unit sphere:
```
MeshVersionFormatted 1
Dimension 3
Vertices
3627
-0.00036388531655804268 -0.59084100230366432 -0.80813289181909509 1
-0.78439279779629811 0.013049560393543505 -0.086235052970709089 1
0.61273104243468524 -0.62122808481889269 0.49072225293415234 1
-0.90280737988265813 0.31834445224612795 -0.29286112375857998 1
0.28432001650314209 0.95857476598535318 0.022223166824128997 1
-0.66789470070486012 -0.37735328981403571 0.64204310053551916 1
0.55637252060673048 -0.81828864337985219 0.1517409229077307 1
-0.11867217128535232 -0.5394778599971175 -0.83489672869820608 1
-0.60235472706253734 0.17161883040865494 -0.78095310556409026 1
-0.93768843842053429 -0.27388111596147013 -0.21886415884693222 1
-0.3604344222866922 -0.16571083402724035 0.91912944679869224 1
0.49870081018930945 -0.54655163996989475 -0.67435213354623591 1
22.92261157108813 6.7550034793181295 5.3780857891462439 -1
-2.2174398966861579 13.238599330095225 -20.489569250762372 -1
16.064966504603031 3.2239623090859375 18.207770820053071 -1
16.911272831483501 -6.1920282801810798 -16.603241761673104 -1
-4.5467602461920604 -14.846036772325373 18.945241181320721 -1
-19.109538703251967 -15.271193885005518 -1.2712859141378501 -1
23.093241057131078 -7.0847138770547939 4.0631326285887948 -1
11.153350650532344 -8.5546255900709589 20.060437439895797 -1
-0.74431913426850649 6.0346304187022763 -23.728236865305334 -1
...
```
Using a different program I was able to visualize them

cgal-ghosts-sphere

When I recompile with
```
> g++ -O3 -frounding-math -DCGAL_LINKED_WITH_TBB -DCGAL_HEADER_ONLY -isystem /path/to/eigen -isystem /path/to/CGAL-5.2.1/include -isystem /path/to/boost_1_74_0/include -isystem /path/to/tbb/include -L/path/to/boost_1_74_0/lib/shared -L/path/to/tbb/lib/ -Wl,-rpath,/path/to/boost_1_74_0/lib/shared -Wl,-rpath,/path/to/tbb/lib/ /path/to/CGAL-5.2/examples/Mesh_3/mesh_implicit_sphere.cpp -lgmp -lmpfr -lboost_program_options -lboost_thread -ltbb -ltbbmalloc -lpthread
```
(the only difference being the removed ```-DCGAL_CONCURRENT_MESH_3```), the problem disappears. Am I setting something obviously wrong in the compilation or is this an issue with how TBB is used in CGAL?

As a side note, the mesh in the first version has
3627 Vertices,
4156 Triangles and
18931 Tetrahedra,
whereas in the second version, there are
3435 Vertices,
4036 Triangles and
17770 Tetrahedra
(i.e., also the number of triangles and tetrahedra changes). Is it possible that the "ghost" vertices in fact belong to some strange inverted elements?

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux 64 bit
* Compiler: g++ 10.2, tested also with g++ 4.8.5
* Release or debug mode: release
* Specific flags used (if any): ```-O3 -frounding-math -DCGAL_CONCURRENT_MESH_3 -DCGAL_LINKED_WITH_TBB -DCGAL_HEADER_ONLY```, see above
* CGAL version: 5.2, tested with 4.14 as well
* Boost version: 1.74, tested with 1.63 as well
* Other libraries versions if used (Eigen, TBB, etc.): Eigen, TBB 2020 update 2

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

This PR documents the function `CGAL::remove_far_points_in_mesh_3()` that removes as a post processing the "far points" inserted to help the concurrent version of the mesher being more efficient.

## Release Management

* Affected package(s): Mesh_3
* Issue(s) solved (if any): fix #5846
* [Small Feature](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Remove_far_points), pre-approved by @lrineau 21 September 2021
* [Link to compiled documentation](https://cgal.github.io/5999/v1/Mesh_3/group__PkgMesh3Functions.html#ga957cd22e1a4a82dc21b9d8659ff936d5)
* License and copyright ownership: unchanged

### CODE REVIEW

**Comment by afabri** [comment]
I would expect the documentation of the  far points in the class  `Mesh_triangulation_3`  where users set the `Concurrency_tag`. And maybe also in the User Manual in the section about parallel algorithms, maybe even with an example.  Instead of remaining vague about the number of inserted points, why don't we say that it is 8 points on the corners of an enlarged bounding box?

**Comment by lrineau** [comment]
> I would expect the documentation of the far points in the class [`Mesh_triangulation_3`](https://cgal.github.io/5999/v1/Mesh_3/structCGAL_1_1Mesh__triangulation__3.html) where users set the `Concurrency_tag`. And maybe also in the User Manual in the section about parallel algorithms, maybe even with an example. Instead of remaining vague about the number of inserted points, why don't we say that it is 8 points on the corners of an enlarged bounding box?
> Also maybe add a word, when they pose a problem. Why should I call the function. I guess we do not call it at the end of each meshing function, because it might be that users call several meshing functions on the same triangulation.

`Mesh_triangulation_3` is just a type generator. We do not want to add anything to that class template. Besides, `Mesh_triangulation_3` is just a shortcut to use Mesh_3, and users can assemble their own version of a triangulation for Mesh_3.

`make_mesh_3`, `refine_mesh_3` are the documented ways to create or modify a c3t3. I find it logical than `remove_far_points_in_mesh_3` is in the same section.

**Comment by lrineau** [comment]
I pre-approve the small feature. My only concern is maybe the name of the function itself. Or we could explain more why the "far points" are named that way.

> **For efficiency reasons**, the concurrent version of the tetrahedral mesh generation algorithm implemented in `CGAL::make_mesh_3()` and `CGAL::refine_mesh_3()` inserts a small set of auxiliary vertices **outside the bounding box of the mesh domain**. **At the that end of the meshing process, they** belong to the triangulation but are isolated from the complex.

File: Mesh_3/include/CGAL/remove_far_points_in_mesh_3.h [hunk 1]
```diff
@@ -71,6 +71,21 @@ class Remove_far_points<C3T3, Parallel_tag>
 
 } // namespace Mesh_3
 
+/*!
+\ingroup PkgMesh3Functions
+
+The concurrent version of the tetrahedral mesh generation algorithm implemented in
+`CGAL::make_mesh_3()` and `CGAL::refine_mesh_3()` inserts a small set of
+auxiliary vertices that belong to the triangulation but are isolated from the complex
+at the end of the meshing process.
```
Comment (lrineau) [inline Mesh_3/include/CGAL/remove_far_points_in_mesh_3.h:80]:
```suggestion
For efficiency reasons, the concurrent version of the tetrahedral mesh generation algorithm implemented in `CGAL::make_mesh_3()` and `CGAL::refine_mesh_3()` inserts a small set of auxiliary vertices outside the bounding box of the mesh domain. At the that end of the meshing process, they belong to the triangulation but are isolated from the complex.
```

**Review by lrineau — APPROVED** [review]
I pre-approve.

**Comment by janetournois** [comment]
This PR is replaced by #6012

**Pull request closed without merging**